server/lib/nl/utils.py

```diff
@@ -143,15 +141,21 @@ def _add_to_set_from_nested_dict(
 
 def remove_stop_words(input_str: str, stop_words: Set[str]) -> str:
   """Remove stop words from a string and return the remaining in lower case."""
-  res = input_str.lower().split()
-  output = ''
-  for w in res:
-    if (w not in stop_words):
-      output += w + " "
-  if not output:
-    return ''
-  else:
-    return output[:-1]
+
+  # Note: we are removing the full sequence of words in every entry in `stop_words`.
+  # For example, if a stop_words entry is "these words remove" then the entire
+  # sequence "these words remove" will potentially be removed and not individual
+  # occurences of "these", "words" and "remove".
+
+  # Creating a leading and trailing space so that "<space>word<space>"
+  # can be matched.
+  input_str = f" {input_str.lower()} "
+  for words in stop_words:
+    # Using regex based replacements.
+    input_str = re.sub(f" {words} ", ' ', input_str)
```

Comment:
Instead of padding spaces, will this prescribed pattern work: https://www.oreilly.com/library/view/regular-expressions-cookbook/9781449327453/ch02s06.html ?

Reply:
super nice suggestion! thanks :) i've incorporated it in a couple of places.